registry/machine.go

```diff
@@ -93,9 +106,78 @@ func (r *EtcdRegistry) SetMachineState(ms machine.MachineState, ttl time.Duratio
 		return uint64(0), err
 	}
 
+	// Set the initial metadata on creation
+	if err = r.setMachineMetadata(ms); err != nil {
+		return uint64(0), err
+	}
+
 	return resp.Node.ModifiedIndex, nil
 }
 
+func (r *EtcdRegistry) UpdateMachineMetadata(machID string, key string, value string) error {
+	//Attempt to update key
+	update := etcd.Update{
```

Comment:
No - SetMachineState does an Update then Set so the event sent to downstream watchers can be used to distinguish a new from an existing machine. It is fine to use Set here, as the event generated by the Set isn't used.